### PR TITLE
BE-docker-compose-for-production

### PR DESCRIPTION
**Pull request by 50-Course**
- :construction: Added healthcheck to `backend` service.
- :beer: Needs improvement on `daphne` conf.

### CODE REVIEW

File: docker-compose.production.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# from which services can extend from;
+#
+# Source: https://nickjanetakis.com/blog/best-practices-around-production-ready-web-apps-with-docker-compose
+x-app: &core
+  build:
+    context: ./backend/
+
+  command: |
+    sh -c "
+    cd ./backend/core &&
```
Comment (Joseph455) [inline docker-compose.production.yml:18]: No need to cd into /backend/core

File: docker-compose.production.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    python manage.py makemigrations &&
```
Comment (Joseph455) [inline docker-compose.production.yml:19]: No need for makemigrations we should0

File: docker-compose.production.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on:
+    - postgres:
+      condition: service-healthy
+  ports:
+    - "${DOCKER_SERVICE_BACKEND_PORT:-8000:8000}"
+  restart:
+    - "${DOCKER_RESTART_POLICY:-unless-stopped}"
+  stop_grace_period: "3s"
+  volumes:
+    - ".:/backend"
```
Comment (Joseph455) [inline docker-compose.production.yml:33]: use volume as - ./backend:/backend

File: docker-compose.production.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    - ./frontend:/frontend
+  networks:
+    - front-net
+    - back-net
+
+
+services:
+  # PREFER Anchors
+  # Source: https://nickjanetakis.com/blog/docker-tip-82-using-yaml-anchors-and-x-properties-in-docker-compose
+  web:
```
Comment (Joseph455) [inline docker-compose.production.yml:55]: change to backend

**Review by Joseph455 — CHANGES_REQUESTED** [review]

File: docker-compose.production.yml (hunk 3 above)
Comment (50-Course) [reply docker-compose.production.yml:33]: Fixed

File: docker-compose.production.yml (hunk 1 above)
Comment (50-Course) [reply docker-compose.production.yml:18]: Fixed

File: docker-compose.production.yml (hunk 2 above)
Comment (50-Course) [reply docker-compose.production.yml:19]: Removed this line

**Pull request merged**